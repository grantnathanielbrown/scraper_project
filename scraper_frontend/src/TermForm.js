import React, { Component } from 'react'

export default class TermForm extends Component {
    render() {
        return (
            <div className="term-form">
                <label>
                Search Keyword 	&nbsp;	&nbsp;	&nbsp;	&nbsp;	&nbsp;       
                <input id="searchTerm" className="rounded search-term" value={this.props.searchTerm} onChange={this.props.handleChange} type="text"/>
                </label>                 
            </div>
        )
    }
}
