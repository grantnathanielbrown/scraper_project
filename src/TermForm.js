import React, { Component } from 'react'

export default class TermForm extends Component {
    render() {
        return (
            <div className="term-form">
                <label>
                Search Keyword 	&nbsp;	&nbsp;	&nbsp;	&nbsp;	&nbsp;       
                <input id="timeframe" className="rounded timeframe" value={this.props.timeframe} onChange={this.props.handleChange} type="text"/>
                </label>

                <label>
                Timeframe 	&nbsp;	&nbsp;	&nbsp;	&nbsp;	&nbsp;       
                <select id="category" onChange={this.props.handleChange}>
                    <option value="Hour">Hour</option>
                    <option value="Day">Day</option>
                    <option value="Week">Week</option>
                    <option value="Month">Month</option>
                    {/* <option value="Random">Random</option> */}
                    <option value="Year">Year</option>
                    <option value="All">All</option>
                </select>
                </label> 

            </div>
        )
    }
}
