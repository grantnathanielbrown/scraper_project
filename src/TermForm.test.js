import React from 'react';
import { shallow } from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import TermForm from './TermForm';


expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('Term Form', () => {
    const mockHandleChange = jest.fn();
    const termForm = shallow(<TermForm handleChange={mockHandleChange} searchTerm='dummy search term' timeframe='dummy time'/>);
    it('renders properly', () => {
        expect(termForm).toMatchSnapshot();
    });

    it('can call the function that is passed down as a prop', () => {
        const sampleSearch = 'cat gifs';
        termForm.find('#searchTerm').simulate('change', {target: {value: sampleSearch}});
        expect(mockHandleChange).toHaveBeenCalledTimes(1);
    });

    
});


