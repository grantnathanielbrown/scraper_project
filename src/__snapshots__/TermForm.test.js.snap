// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Term Form renders properly 1`] = `
<div
  className="term-form"
>
  <label>
    Search Keyword           
    <input
      className="rounded"
      id="searchTerm"
      onChange={[MockFunction]}
      type="text"
      value="dummy search term"
    />
  </label>
  <label>
    Timeframe           
    <select
      id="category"
      onChange={[MockFunction]}
    >
      <option
        value="Hour"
      >
        Hour
      </option>
      <option
        value="Day"
      >
        Day
      </option>
      <option
        value="Week"
      >
        Week
      </option>
      <option
        value="Month"
      >
        Month
      </option>
      <option
        value="Year"
      >
        Year
      </option>
      <option
        value="All"
      >
        All
      </option>
    </select>
  </label>
</div>
`;
